internal/cmd/agent/operator.go

```diff
@@ -151,6 +152,12 @@ func newReconciler(ctx, localCtx context.Context, mgr manager.Manager, localConf
 	}
 	localClient := localCluster.GetClient()
 
+	kubeconfig := flag.Lookup("kubeconfig").Value.String()
+	if kubeconfig != "" {
+		// set KUBECONFIG env var so helm can find it
+		os.Setenv("KUBECONFIG", flag.Lookup("kubeconfig").Value.String())
```

Comment:
Why do the lookup again?
```suggestion
		os.Setenv("KUBECONFIG", kubeconfig)
```